inference-engine/src/cldnn_engine/cldnn_graph.cpp

```diff
@@ -667,24 +673,26 @@ void CLDNNGraph::GetPerformanceCounts(std::map<std::string, InferenceEngine::Inf
         bool existInProfiling = std::find(profilingIDs.begin(), profilingIDs.end(), primId) != profilingIDs.end();
         if ((!existInProfiling || (existInProfiling && perfIter->second.first.length() == 0)) &&
             executedPrimitives.find(primId) != executedPrimitives.end()) {
-            auto event = executedPrimitives.at(primId);
-
-            cldnn::instrumentation::profiling_info cldnnInfo{primId, event.get_profiling_info()};
+            auto event_list = executedPrimitives.at(primId);
 
             // Collect timings
             long long cpuTime = 0;
             long long deviceTime = 0;
 
-            for (auto &interval : cldnnInfo.intervals) {
-                using duration_t = std::chrono::duration<long long, std::chrono::microseconds::period>;
-                auto count = std::chrono::duration_cast<duration_t>(interval.value->value()).count();
+            for (size_t k = 0; k < event_list.get_events().size(); k++) {
+                cldnn::instrumentation::profiling_info kernelEventInfo{primId, event_list.get_events()[k].get_profiling_info()};
+
+                for (auto &interval : kernelEventInfo.intervals) {
+                    using duration_t = std::chrono::duration<long long, std::chrono::microseconds::period>;
+                    auto count = std::chrono::duration_cast<duration_t>(interval.value->value()).count();
 
-                if (interval.name == "submission") {
-                    cpuTime += count;
-                } else if (interval.name == "executing") {
-                    deviceTime += count;
-                } else if (interval.name == "duration") {  // "duration" is used for CPU layers
-                    cpuTime += count;
+                    if (interval.name == "submission") {
+                        cpuTime += count;
+                    } else if (interval.name == "executing") {
+                        deviceTime += count;
```

Comment:
This won't be correct if the events correspond to commands executed in parallel.

Reply:
Fixed